plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergParquetWithBloomFilters.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.plugin.hive.TestingHivePlugin;
+import io.trino.testing.BaseTestParquetWithBloomFilters;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import static io.trino.plugin.hive.parquet.TestHiveParquetWithBloomFilters.writeParquetBloomFilterSource;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static java.lang.String.format;
+
+public class TestIcebergParquetWithBloomFilters
+        extends BaseTestParquetWithBloomFilters
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = IcebergQueryRunner.builder().build();
+        dataDirectory = queryRunner.getCoordinator().getBaseDataDir().resolve("iceberg_data");
+
+        // create hive catalog
+        queryRunner.installPlugin(new TestingHivePlugin());
+        queryRunner.createCatalog("hive", "hive", ImmutableMap.<String, String>builder()
+                .put("hive.metastore", "file")
+                .put("hive.metastore.catalog.dir", dataDirectory.toString())
+                .put("hive.security", "allow-all")
+                .buildOrThrow());
+
+        return queryRunner;
+    }
+
+    @Override
+    protected String createParquetTableWithBloomFilter(String columnName, List<Integer> testValues)
+    {
+        // create the managed table
+        String tableName = "parquet_with_bloom_filters_" + randomNameSuffix();
+        String hiveTableName = format("hive.tpch.%s", tableName);
+        String icebergTableName = format("iceberg.tpch.%s", tableName);
+        assertUpdate(
+                format(
+                        "CREATE TABLE %s (%s INT) WITH (format = 'PARQUET')",
+                        hiveTableName,
+                        columnName));
+
+        // directly write data to the managed table
+        Path tableLocation = Path.of("%s/tpch/%s".formatted(dataDirectory, tableName));
+        Path fileLocation = tableLocation.resolve("bloomFilterFile.parquet");
+        writeParquetBloomFilterSource(fileLocation.toFile(), columnName, testValues);
+
+        // migrate the hive table to the iceberg table
```

Comment:
@ebyhr @findinpath do you know if there is a simpler way to create an iceberg table from an existing parquet file than to create table in hive and then migrate it to iceberg ?

Reply:
For this unit the hive catalog and iceberg catalog shared the same URL, thus I could use Hive writer to generate the data and then migrate the hive table to the iceberg table. 

more details at: https://github.com/apache/iceberg/issues/2068. Summary is Iceberg has strong specification for existing data. Once the bloomFilter write for the optimized parquet writer is done, we can directly create an iceberg table containing the parquet files with bloomfilters.